adam-core/src/main/scala/org/bdgenomics/adam/rdd/contig/NucleotideContigFragmentRDDFunctions.scala

```diff
@@ -49,6 +51,59 @@ class NucleotideContigFragmentRDDFunctions(rdd: RDD[NucleotideContigFragment]) e
   }
 
   /**
+   * Save nucleotide contig fragments in FASTA format.
+   *
+   * @param fileName file name
+   * @param lineWidth hard wrap FASTA formatted sequence at line width, default 60
+   */
+  def saveAsFasta(fileName: String, lineWidth: Int = 60) = {
+
+    def isFragment(record: NucleotideContigFragment): Boolean = {
+      Option(record.fragmentNumber).isDefined && Option(record.numberOfFragmentsInContig).fold(false)(_ > 1)
+    }
+
+    def toFasta(record: NucleotideContigFragment): String = {
+      val sb = new StringBuilder()
+      sb.append(">")
+      sb.append(record.contig.contigName)
+      Option(record.description).foreach(n => sb.append(" ").append(n))
+      if (isFragment(record)) {
+        sb.append(" fragment %d of %d".format(record.fragmentNumber + 1, record.numberOfFragmentsInContig))
+      }
+      for (line <- Splitter.fixedLength(lineWidth).split(record.fragmentSequence)) {
+        sb.append("\n")
+        sb.append(line)
+      }
+      sb.toString
+    }
+
+    rdd.map(toFasta).saveAsTextFile(fileName)
+  }
+
+  /**
+   * Merge fragments by contig name.
+   */
+  def mergeFragments(): RDD[NucleotideContigFragment] = {
+
+    def merge(first: NucleotideContigFragment, second: NucleotideContigFragment): NucleotideContigFragment = {
+      val merged = NucleotideContigFragment.newBuilder(first)
+        .setFragmentNumber(null)
+        .setFragmentStartPosition(null)
+        .setNumberOfFragmentsInContig(null)
+        .setFragmentSequence(first.fragmentSequence + second.fragmentSequence)
+        .build
+
+      merged
+    }
+
+    rdd
+      .sortBy(fragment => (fragment.contig.contigName, Option(fragment.fragmentNumber) match { case Some(n) => n.toInt case None => -1 }))
```

Comment:
I might do `Option(fragment.fragmentNumber).map(_.toInt).getOrElse(-1)` here, fwiw. If that doesn't seem cleaner then feel free to ignore.